src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestBindCallGen.generated.txt

```diff
@@ -0,0 +1,122 @@
+// <auto-generated/>
+#nullable enable
+
+using System.Linq;
+
+internal static class GeneratedConfigurationBinder
+{
+    public static void Bind(this global::Microsoft.Extensions.Configuration.IConfiguration configuration, Program.MyClass obj) => BindCore(configuration, ref obj);
+
+    private static void BindCore(global::Microsoft.Extensions.Configuration.IConfiguration configuration, ref Program.MyClass obj)
+    {
+        if (obj is null)
+        {
+            throw new global::System.ArgumentNullException(nameof(obj));
+        }
+
+        if (configuration["MyString"] is string stringValue0)
+        {
+            obj.MyString = stringValue0;
+        }
+
+        if (configuration["MyInt"] is string stringValue1)
+        {
+            obj.MyInt = int.Parse(stringValue1);
+        }
+
+        System.Collections.Generic.List<int> temp2 = obj.MyList;
+        temp2 ??= new System.Collections.Generic.List<int>();
+        BindCore(configuration.GetSection("MyList"), ref temp2);
+        obj.MyList = temp2;
+
+        System.Collections.Generic.Dictionary<string, string> temp3 = obj.MyDictionary;
+        temp3 ??= new System.Collections.Generic.Dictionary<string, string>();
```

Comment:
>Dictionary [](http://example.com/codeflow?start=49&length=10)

do we have a test case for dictionary using integer or Enum keys?

Reply:
Yes we have a few in `ConfigurationBinderTests.cs`.